webcrypto/subtle_crypto.go

```diff
@@ -218,9 +217,86 @@ func (sc *SubtleCrypto) Decrypt(algorithm, key, data goja.Value) *goja.Promise {
 // `algorithm` identifies a public-key cryptosystem, this is the private key.
 //
 // The `data` parameter should contain the data to be signed.
-func (sc *SubtleCrypto) Sign(algorithm goja.Value, key goja.Value, data []byte) *goja.Promise {
-	// TODO: implementation
-	return nil
+func (sc *SubtleCrypto) Sign(algorithm, key, data goja.Value) *goja.Promise {
+	rt := sc.vu.Runtime()
+	promise, resolve, reject := sc.makeHandledPromise()
+
+	// 2.
+	// We obtain a copy of the key data, because we might need to modify it.
+	ab, err := exportArrayBuffer(rt, data)
+	if err != nil {
+		reject(err)
+		return promise
+	}
+	dataToSign := make([]byte, len(ab))
+	copy(dataToSign, ab)
+
+	// 3.
+	normalized, err := normalizeAlgorithm(rt, algorithm, OperationIdentifierSign)
+	if err != nil {
+		reject(err)
+		return promise
+	}
+
+	var ck CryptoKey
+	if err = rt.ExportTo(key, &ck); err != nil {
+		reject(NewError(0, InvalidAccessError, "key argument does hold not a valid CryptoKey object"))
+		return promise
+	}
+
+	keyAlgorithmNameValue, err := traverseObject(rt, key.ToObject(rt), "algorithm", "name")
+	if err != nil {
+		reject(err)
+		return promise
+	}
+
+	go func() {
+		// 8.
+		if normalized.Name != keyAlgorithmNameValue.String() {
+			reject(NewError(0, InvalidAccessError, "algorithm name does not match key algorithm name"))
+			return
+		}
+
+		// 9.
+		for !containsUsage(ck.Usages, SignCryptoKeyUsage) {
```

Comment:
Should we move `containsUsage` function on the crypto key?
```suggestion
		for !ck.containsUsage(SignCryptoKeyUsage) {
```